### ISSUE TITLE
False positive for `Layout/RedundantLineBreak` with single newline in quoted symbol

### ISSUE THREAD

**Issue by ymap** [rubocop/rubocop#12192]
I've noticed a peculiar behavior with the `Layout/RedundantLineBreak` cop. It seems to have an issue when encountering a quoted symbol that concludes with a single newline. I acknowledge this is a niche scenario, not something one would run into often, but thought it should be highlighted.

Here's a concise reproducible example:

```ruby
foo(:"
")
```

---

## Expected behavior

No offenses should be reported.

## Actual behavior

An offense is reported. When using `--autocorrect`, the code is modified to:

```ruby
foo(:" ")
```

## Steps to reproduce the problem

1. Save the provided code in a file named `example.rb`.
2. Run `bundle exec rubocop --only Layout/RedundantLineBreak example.rb`.

## RuboCop version

```
$ bundle exec rubocop -V
1.56.2 (using Parser [3.2.2.3](http://3.2.2.3/), rubocop-ast 1.29.0, running on ruby 3.0.2) [x86_64-linux]
  - rubocop-performance 1.19.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.2
```
